### PR TITLE
Add dump/parse support for Font

### PR DESCRIPTION
**Pull request by GEEKiDoS**

### CODE REVIEW

**Comment by RektInator** [comment]
The PR is not bad, but:
* I want to step away from the #define macros per asset dump code, this is from a legacy version of zonetool and can probably be implemented in a cleaner way
* Code duplication, font assets should be the same between at least IW4 and IW5, you can simply call the IW5 dump / parse functions. If struct changes are required, you can transform the struct in the IW4 Dump function first and then pass the transformed struct to the dump function. You can then apply the opposite logic after calling the IW5 parse function.

**Comment by GEEKiDoS** [comment]
> The PR is not bad, but:
> 
> * I want to step away from the #define macros per asset dump code, this is from a legacy version of zonetool and can probably be implemented in a cleaner way
> * Code duplication, font assets should be the same between at least IW4 and IW5, you can simply call the IW5 dump / parse functions. If struct changes are required, you can transform the struct in the IW4 Dump function first and then pass the transformed struct to the dump function. You can then apply the opposite logic after calling the IW5 parse function.

I see, I'll fix this

**Comment by RektInator** [comment]
Can you merge the latest develop into your branch so that the code can be compiled by the CI?

File: src/IW4/IW4.cpp [hunk 1]
```diff
@@ -293,6 +293,7 @@ char**>(0x00799278)[type]);
 					DECLARE_ASSET(lightdef, ILightDef);
 					DECLARE_ASSET(weapon, IWeaponDef);
 					DECLARE_ASSET(addon_map_ents, IAddonMapEnts);
+					DECLARE_ASSET(font, IFontDef);
```
Comment (RektInator) [inline src/IW4/IW4.cpp:296]: Do this on IW5 too please :P

Comment (GEEKiDoS) [reply src/IW4/IW4.cpp:296]: My bad

**Pull request merged**